pkg/health/types.go

```diff
@@ -91,22 +140,35 @@ func (h *Health) WithDetails(details map[string]interface{}) *Health {
 type Indicator interface {
 	// Name returns the name of the component
 	Name() string
-	// Health returns the health of the component
-	Health() *Health
+
+	// Interval returns settings of the indicator
+	Interval() time.Duration
+
+	// FailuresTreshold returns the maximum failures in a row until component is considered down
+	FailuresTreshold() int64
+
+	// Fatal returns if the health indicator is fatal for the overall status
+	Fatal() bool
+
+	// Status returns the health information of the component
+	Status() (interface{}, error)
 }
 
-// AggregationPolicy is an interface to provide aggregated health information
-//go:generate counterfeiter . AggregationPolicy
-type AggregationPolicy interface {
-	// Apply processes the given healths to build a single health
-	Apply(healths map[string]*Health) *Health
+// ConfigurableIndicator is an interface to provide configurable health of a component
+//go:generate counterfeiter . ConfigurableIndicator
+type ConfigurableIndicator interface {
```

Comment:
I would suggest tey to get rid of this interface. - thereare some ideas in the other comments